x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/index.tsx

```diff
@@ -277,3 +303,127 @@ export const StatefulEditDataProvider = React.memo<Props>(
 );
 
 StatefulEditDataProvider.displayName = 'StatefulEditDataProvider';
+
+function isStringOrNumberArray(
```

Comment:
Given the following unit tests:

```ts
  describe('isStringOrNumberArray', () => {
    test('it returns false when value is not an array', () => {
      expect(isStringOrNumberArray('just a string')).toBe(false);
    });

    test('it returns false when value is an array of mixed types', () => {
      expect(isStringOrNumberArray(['mixed', 123, 'types', {}, undefined, null])).toBe(false);
    });

    test('it returns true when value is an empty array', () => {
      expect(isStringOrNumberArray([])).toBe(true);
    });

    test('it returns true when value is an array of all strings', () => {
      expect(isStringOrNumberArray(['all', 'string', 'values'])).toBe(true);
    });

    test('it returns true when value is an array of all numbers', () => {
      expect(isStringOrNumberArray([123, 456, 789])).toBe(true);
    });
  });
```

The current implementation fails some tests:

```
    isStringOrNumberArray
      ✓ it returns false when value is not an array
      ✕ it returns false when value is an array of mixed types (1 ms)
      ✕ it returns true when value is an empty array
      ✓ it returns true when value is an array of all strings
      ✓ it returns true when value is an array of all numbers
```

Consider something like the following changes to address the test failures:

```ts
export const isStringOrNumberArray = (value: unknown): value is Array<string | number> =>
  Array.isArray(value) &&
  (value.length === 0 ||
    value.every((x) => typeof x === 'string') ||
    value.every((x) => typeof x === 'number'));
```

Would you be willing to adopt the unit tests above?

Reply:
I have but left out the second check:
 value.every returns true for an empty array:
 https://stackoverflow.com/a/34137289